### DISCUSSION TITLE
GitHub code checking

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2229] [Questions & Answers] [answered]
As you know every source code in a PR is analyzed by GitHub bots and if there are errors they need to be fixed. Has anyone done in his own repository the following experiment so far to give all the current OpenMage code in a PR and check for errors?

**Comment by Flyingmana** [comment]
you need to be a bit more specific here. Do you have examples for these checks?

**Reply by addison74** [reply]
The idea is to determine GitHub bots analyze all existing OpenMage code. In order to achieve this task I have to create a PR to be merged. Before merging it GitHub will check the whole code. So far the analysis has been done by PR which has solved pieces of the original Magento code and which today is called OpenMage but the code as a whole has never been analyzed. It can be done in an IDE like PHPStorm, but I was also interested to see what GitHub thinks.

**Reply by fballiano** [reply]
If I'm not mistaken all the code is always checked, not only the changed files of a PR.

**Comment by elidrissidev** [comment, marked as answer]
It does check the whole codebase already. If you remember, checks were failing for new PRs some time ago when PHPStan detected violations originating from old PRs that were merged.
